state/watcher.go

```diff
@@ -505,6 +505,20 @@ func (st *State) WatchModelMachines() StringsWatcher {
 	return newLifecycleWatcher(st, machinesC, members, filter, nil)
 }
 
+// WatchModelMachinesLXDProfile returns a StringsWatcher that notifies
+// when a charm upgrade requests a profile upgrade too.
+func (st *State) WatchModelMachinesLXDProfile() StringsWatcher {
+	members := bson.D{{"upgradelxdprofileapplication", bson.D{{"$regex", "^[A-Za-z]"}}}}
+	filter := func(id interface{}) bool {
+		k, err := st.strictLocalID(id.(string))
+		if err != nil {
+			return false
+		}
+		return !strings.Contains(k, "/")
+	}
+	return newLifecycleWatcher(st, machinesC, members, filter, nil)
```

Comment:
a LifecycleWatcher is only supposed to track what units go from not existing to being Alive, to Dying to Dead. It isn't supposed to tell you when something changes on a given machine document (IIRC).
I think you need a different style of watcher if you want to know when a doc changes an attribute other than Life.